### PR TITLE
use vector of UP as MultiSearch::Children

### PR DESCRIPTION
**Pull request by arnej27959**
* add helper class for constructing MultiSearch::Children
  (mostly for unit tests)
* rewrite as needed to adapt

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@havardpe @geirst @toregge please review

### CODE REVIEW

**Review by havardpe — APPROVED** [review]

File: searchlib/src/vespa/searchlib/queryeval/sourceblendersearch.cpp [hunk 1]
```diff
@@ -158,14 +158,14 @@ SourceBlenderSearch::setChild(size_t index, SearchIterator::UP child) {
...
 }
 
-SourceBlenderSearch *
+SearchIterator::UP
 SourceBlenderSearch::create(std::unique_ptr<sourceselector::Iterator> sourceSelector,
                             const Children &children, bool strict)
 {
     if (strict) {
-        return new SourceBlenderSearchStrict(std::move(sourceSelector), children);
+        return SearchIterator::UP(new SourceBlenderSearchStrict(std::move(sourceSelector), children));
```
Comment (baldersheim) [inline searchlib/src/vespa/searchlib/queryeval/sourceblendersearch.cpp:166]: Should you not use std::make_unique instead ?

**Review by baldersheim — APPROVED** [review]
This is a long awaited modernisation. I just had one comment on SearchIterator::UP versus std::make_unique.

File: searchlib/src/vespa/searchlib/queryeval/sourceblendersearch.cpp (hunk 1 above)
Comment (arnej27959) [reply searchlib/src/vespa/searchlib/queryeval/sourceblendersearch.cpp]: I wanted to handle Strict and non-Strict the same way, and the other constructor wasn't public; will adjust with a NonStrict variant and use make_unique.

**Review by geirst — COMMENTED** [review]
👍

**Pull request merged**